google/cloud/firestore_v1/async_vector_query.py

```diff
@@ -65,49 +74,83 @@ async def get(
                 should be retried.  Defaults to a system-specified policy.
             timeout (float): The timeout for this request.  Defaults to a
                 system-specified value.
+            explain_options
+                (Optional[:class:`~google.cloud.firestore_v1.query_profile.ExplainOptions`]):
+                Options to enable query profiling for this query. When set,
+                explain_metrics will be available on the returned generator.
 
         Returns:
-            list: The vector query results.
+            QueryResultsList[DocumentSnapshot]: The documents in the collection
+            that match this query.
         """
+        explain_metrics: ExplainMetrics | None = None
+
         stream_result = self.stream(
-            transaction=transaction, retry=retry, timeout=timeout
+            transaction=transaction,
+            retry=retry,
+            timeout=timeout,
+            explain_options=explain_options,
         )
         result = [snapshot async for snapshot in stream_result]
-        return result  # type: ignore
 
-    async def stream(
+        if explain_options is None:
+            explain_metrics = None
+        else:
+            explain_metrics = await stream_result.get_explain_metrics()
+
+        return QueryResultsList(result, explain_options, explain_metrics)
+
+    async def _make_stream(
         self,
-        transaction=None,
-        retry: retries.AsyncRetry = gapic_v1.method.DEFAULT,
+        transaction: Optional[transaction.Transaction] = None,
+        retry: Optional[retries.Retry] = gapic_v1.method.DEFAULT,
         timeout: Optional[float] = None,
-    ) -> AsyncGenerator[async_document.DocumentSnapshot, None]:
-        """Reads the documents in the collection that match this query.
+        explain_options: Optional[ExplainOptions] = None,
+    ) -> AsyncGenerator[[DocumentSnapshot | query_profile_pb.ExplainMetrics], Any]:
+        """Internal method for stream(). Read the documents in the collection
+        that match this query.
 
-        This sends a ``RunQuery`` RPC and then returns an iterator which
+        This sends a ``RunQuery`` RPC and then returns a generator which
         consumes each document returned in the stream of ``RunQueryResponse``
         messages.
 
+        .. note::
+
+           The underlying stream of responses will time out after
+           the ``max_rpc_timeout_millis`` value set in the GAPIC
+           client configuration for the ``RunQuery`` API.  Snapshots
+           not consumed from the iterator before that point will be lost.
```

Comment:
is there any context behind this new comment?

Reply:
I think I just copied from other `_make_stream()` docstrings together with the rest of the docstring, see [all occurrences](https://github.com/search?q=repo%3Agoogleapis%2Fpython-firestore%20%22max_rpc_timeout_millis%22&type=code). They seem to only appear for query and collection - does it apply to vector queries?

Reply:
Ok, I don't really think it applies to async code, but it's an internal method so I guess I could take it or leave it for now. I can look closer at this comment when we refactor this later

Reply:
Sounds good, I will delete it because I'm not even sure if it applies here.

Reply:
done